src/main/java/com/coveo/spillway/SpillwayFactory.java

```diff
@@ -58,9 +60,18 @@ public SpillwayFactory(LimitUsageStorage storage, Clock clock) {
    * @param resource The name of the resource on which the limit are enforced
    * @param limits The different enforced limits
    * @return The new {@link Spillway}
+   * @throws SpillwayLimitWithSameNameException
    */
   @SafeVarargs
-  public final <T> Spillway<T> enforce(String resource, Limit<T>... limits) {
+  public final <T> Spillway<T> enforce(String resource, Limit<T>... limits)
+      throws SpillwayLimitWithSameNameException {
+    int numberOfDistinctLimits =
+        (int) Arrays.stream(limits).map(limit -> limit.getName()).distinct().count();
+
+    if (numberOfDistinctLimits != Arrays.asList(limits).size()) {
```

Comment:
You cannot chose which limit to verify against when you receive the result of the addAndGet ;)